lib/experimental/class-wp-nav-menu-conversion-gutenberg.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * WP_Nav_Menu_Conversion_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages Fallback behavior for Navigation menus.
+ *
+ * @access public
+ */
+class WP_Nav_Menu_Conversion_Gutenberg {
+
+	/**
+	 * The Menu term object of the menu to convert.
+	 *
+	 * @var WP_Term
+	 */
+	private $menu;
+
+	/**
+	 * Constructor
+	 *
+	 * @param WP_Term $menu The Menu term object of the menu to convert.
+	 */
+	public function __construct( $menu ) {
+		$this->menu = $menu;
+	}
+
+	/**
+	 * Converts a Classic Menu to blocks.
+	 *
+	 * @return string the serialized and normalized parsed blocks.
+	 */
+	public function convert() {
+		$menu_items = wp_get_nav_menu_items( $this->menu->term_id, array( 'update_post_term_cache' => false ) );
+
+		// Set up the $menu_item variables.
+		// Adds the class property classes for the current context, if applicable.
+		_wp_menu_item_classes_by_context( $menu_items );
+
+		$menu_items_by_parent_id = $this->group_by_parent_id( $menu_items );
+
+		$inner_blocks = $this->to_blocks(
+			isset( $menu_items_by_parent_id[0] )
```

Comment:
I think we should move this conditional out of the methof call and make a new variable, it's hard to read as is (I think that's how it was in the block renderer)

Reply:
Ok something like `$first_menu_item`?

Reply:
✅